src/bgw/job_stat.c

```diff
@@ -205,6 +205,18 @@ calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failure
 	float8 multiplier = (consecutive_failures > MAX_FAILURES_MULTIPLIER ? MAX_FAILURES_MULTIPLIER :
 																		  consecutive_failures);
 	MemoryContext oldctx;
+
+	/* stop to execute failing jobs after reached the "max_retries" option */
+	if (job->fd.max_retries > 0 && consecutive_failures >= job->fd.max_retries)
+	{
+		elog(LOG,
+			 "job %d not executed as max_retries reached %d, consecutive failures %d",
+			 job->fd.id,
+			 job->fd.max_retries,
+			 consecutive_failures);
+		return res;
```

Comment:
I would actually suggest to make this (at least) a warning and use `ereport`.

I think it should be a warning since it is something that potentially is a problem and users will look for warnings and errors in the log to figure out "what is wrong". If you look at [Message Severity Levels](https://www.postgresql.org/docs/14/runtime-config-logging.html#RUNTIME-CONFIG-SEVERITY-LEVELS), `LOG` level "Reports information of interest to administrators, e.g., checkpoint activity", while `WARNING` level "Provides warnings of likely problems, e.g., COMMIT outside a transaction block." We could actually consider `ERROR` as well since it says "Reports an error that caused the current command to abort."

I think that you should use `ereport` because some of that information belongs in the error details and is not necessarily interesting to log. For example, if you look in `acl.c` it has an `ereport` that report the failure as a brief message and provide more details in the `errdetail`. So what job number failed makes sense to have in the message, but the number of retries and the maximum belong in the `errdetail`.

```
			ereport(ERROR,
					(errcode(ERRCODE_NAME_TOO_LONG),
					 errmsg("identifier too long"),
					 errdetail("Identifier must be less than %d characters.",
							   NAMEDATALEN)));
```

Reply:
Makes sense. Will change to use `ereport`. Now I discovered a bug in the logic leading to log forever the failing job